A33-3p-identity.md

```diff
@@ -0,0 +1,109 @@
+
+### Related Proposals: 
+N/A
+
+## Proposal
+
+This doc proposes to extend the access to Google APIS directly with 3p identities. The crednetials of the 3p identities with be in one of the 3 following formats:
+
+**File-source credentials**
+```
+{
+  "type": "external_account",
+  "audience": "//iam.googleapis.com/project/$PROJECT_NUMBER/locations/global/workloadIdentityPools/$POOL_ID/providers/$PROVIDER_ID",
+  "subject_token_type": "urn:ietf:params:oauth:token-type:jwt",
+  "service_account_impersonation_url": "https://iamcredentials.googleapis.com/v1/projects/-/serviceAccounts/$EMAIL:generateAccessToken",
+  "token_url": "https://sts.googleapis.com/v1beta/token",
+  "token_info_url": "https://sts.googleapis.com/v1alpha/token_info",
+  "credential_source": {
+    "file": "/var/run/secrets/goog.id/token"
+  }
+}
+```
+
+**Url-sourced credentials**
+```
+{
+  "type": "external_account",
+  "audience": "//iam.googleapis.com/project/$PROJECT_NUMBER/locations/global/workloadIdentityPools/$POOL_ID/providers/$PROVIDER_ID",
+  "subject_token_type": "urn:ietf:params:oauth:token-type:jwt",
+  "service_account_impersonation_url": "https://iamcredentials.googleapis.com/v1/projects/-/serviceAccounts/$EMAIL:generateAccessToken",
+  "token_url": "https://sts.googleapis.com/v1beta/token",
+  "token_info_url": "https://sts.googleapis.com/v1alpha/token_info",
+  "credential_source": {
+    "headers": {
+      "Metadata": "True"
+    },
+    "url": "http://169.254.169.254/metadata/identity/oauth2/token?api-version=2018-02-01&resource=https://iam.googleapis.com/project/$PROJECT_NUMBER/locations/global/workloadIdentityPools/$POOL_ID/providers/$PROVIDER_ID"
+  }
+}
+```
+
+**AWS credentials**
+```
+{
+  "type": "external_account",
+  "audience": "//iam.googleapis.com/project/$PROJECT_NUMBER/locations/global/workloadIdentityPools/$POOL_ID/providers/"$PROVIDER_ID",
+  "subject_token_type": "urn:ietf:params:aws:token-type:aws4_request",
+  "service_account_impersonation_url": "https://iamcredentials.googleapis.com/v1/projects/-/serviceAccounts/$EMAIL:generateAccessToken",
+  "token_url": "https://sts.googleapis.com/v1beta/token",
+  "token_info_url": "https://sts.googleapis.com/v1alpha/token_info",
+  "credential_source": {
+    "environment_id": "aws1",
+    "region_url": "http://169.254.169.254/latest/meta-data/placement/availability-zone",
+    "url": "http://169.254.169.254/latest/meta-data/iam/security-credentials",
+    "regional_cred_verification_url": "https://sts.{region}.amazonaws.com?Action=GetCallerIdentity&Version=2011-06-15",
+    "cred_verification_url": "https://sts.amazonaws.com?Action=GetCallerIdentity&Version=2011-06-15"
+  }
+}
+```
+
+**Implicit flow**
+
+In core, the underlying credentials retrieval process of `grpc::GoogleDefaultCredentials()` will be extended to accept 3p identity credentials implicitly.
```

Comment:
How does `GoogleDefaultCredentials()` find the above JSON input?  Please document.